rewrites/src/main/scala/fix/scala213/ExplicitNonNullaryApply.scala

```diff
@@ -0,0 +1,38 @@
+package fix.scala213
+
+import scala.PartialFunction.{ cond, condOpt }
+import scala.collection.mutable
+
+import scala.meta._
+
+import scalafix.v1._
+
+final class ExplicitNonNullaryApply extends SemanticRule("fix.scala213.ExplicitNonNullaryApply") {
+  override def fix(implicit doc: SemanticDocument) = {
+    val handled = mutable.Set.empty[Name]
+
+    def fix(tree: Tree, meth: Term, noTypeArgs: Boolean, noArgs: Boolean) = {
+      for {
+        name <- termName(meth)
+        if handled.add(name)
+        if noArgs
+        if name.isReference
+        if !name.parent.exists(_.is[Term.ApplyInfix])
+        info <- name.symbol.info
+        if !info.isJava
+        if cond(info.signature) { case MethodSignature(_, List(Nil), _) => true }
```

Comment:
Does this handle the case when Scala nullary methods override non-nullary methods? When I tried to implement this rule I hit on a wall where the rewrite inserted parentheses in this scenario when the expected case was to keep no parentheses because the Scala method is defined as nullary. When I looked into it, the Scala signatures did not distinguish between non-nullary methods that override non-nullary methods and non-nullary Scala methods without overrides. See https://github.com/scala/bug/issues/10480

Reply:
Every Java method with an empty parameter list is has non-nullary signatures so the inserted parentheses came in up a lot of unwanted cases.

Reply:
Probably not, but thanks for the reminder - I'd forgotten the overriding variants.

Ugh, so many variants.  I'd love if there were just a tree-to-tree transform...  Or a way to generate the variations to test them all out...

Reply:
It's a really complication story, but I've been pushing forward on it.  Basically nullary (`Nil`) overriding nilary (`List(Nil)`) methods, and vice-versa, are going away - except for anything that's defined in Java (where you can override anyway you want and call any way you want.  It'll be a warning in 2.13 (probably 2.13.3, not the next 2.13.2 release) and error in 3.0: https://github.com/scala/scala/pull/8846.

Due to missing info, for now this rewrite will insert parens in (1) almost all the cases with methods defined in Java (or universal) 😞, see `ExplicitNonNullaryApplyJava`, and (2) all 4 cases of overriding in Scala:

```scala
  class Meth { def d() = "" }
  class Prop { def d   = "" }

  class Meth2Prop extends Meth { override def d   = "" }
  class Prop2Meth extends Prop { override def d() = "" }

  val meth2prop = new Meth2Prop
  val prop2meth = new Prop2Meth
```

```diff
   meth2prop.d()
-  meth2prop.d // <- should call with parens
+  meth2prop.d() // <- should call with parens
   prop2meth.d()
-  prop2meth.d // <- should call with parens
+  prop2meth.d() // <- should call with parens
```

If you can spot any way to do it better or even breadcrumbs to how I could make it better, that would be appreciated.

Also, would you be happy to fold these into Scalafix proper?  To make it even more accessible for Scala users (like the procedure syntax one is there).